qemu/tests/boot_with_different_vectors.py

```diff
@@ -45,41 +49,192 @@ def boot_guest_with_vectors(vectors):
         vm.verify_alive()
         return vm
 
+    def get_MSI_queue_from_traceview_output(output):
+        """
+        Extract msi&queues infomation from traceview log file output
+
+        :param output: the content of traceview processed log infomation
+        """
+
+        def _convert_fun(item):
+            value_pair = list(reversed(item.split()))
+            if not value_pair[1].isdigit():
+                value_pair[0] = 'MSIs'
+                value_pair[1] = 0
+            else:
+                value_pair[1] = int(value_pair[1])
+            return value_pair
+
+        info_str = "Start checking dump content for msi & queues info"
+        error_context.context(info_str, logging.info)
+        search_exp = r"\d+ MSIs|\d+ queues"
+        lines = output.split('\n')
+        for line in lines:
+            # special case for vectors = 0
+            queue_when_no_msi = re.search(r'No MSIX, using (\d+) queue', line)
+            if queue_when_no_msi:
+                return (0, int(queue_when_no_msi.group(1)))
+
+            search_res = re.findall(search_exp, line)
+            if not search_res:
+                continue
+            try:
+                value_dict = dict(map(_convert_fun, search_res))
+                MSIs_number = value_dict["MSIs"]
+                queues_number = value_dict["queues"]
+                return (MSIs_number, queues_number)
+            except Exception:
+                pass
+        return (None, None)
+
+    def get_MSIs_and_queues_windows():
+        """
+        Get msi&queues infomation of currentwindows guest.
+        First start a traceview session, then restart the nic interface
+        to trigger logging. By analyzing the dumped output, the msi& queue
+        info is acquired.
+
+        :return: a tuple of (msis, queues)
+        """
+
+        session_serial = vm.wait_for_serial_login(login_timeout)
+        # start traceview
+        error_context.context("Start trace view with pdb file", logging.info)
+        log_path = "c:\\logfile.etl"
```

Comment:
Some parts of the code can be reused in this function. Eg:
```python
def _remove_file(file_path):
         clean_cmd = "del %s" % file_path
         session_serial.cmd(clean_cmd)
```
You can define some small function like above to make the code easier to read, and reduce string formats. Some actions are symmetry, functions would help a little bit.

But the current implantation is OK to me.